docs/proposals/validation-tools/readme.md

```diff
@@ -0,0 +1,83 @@
+
+The current user experience of the linter vs code extension is very simple and straight forward and we&#39;re planning on keeping this experience and extend it to cover other similar tools such as the semantic validator. To summarize it:
+
+- The user opens VS code and creates a new Swagger or edits an existing one.
+- As he';s typing, in the background the SVT extension&#39;s client will be calling the extension&#39;s server (which is basically Autorest) and uploading the updated swagger content.
+- The swagger server will trigger the configured and registered tools for &quot;only the fly validation&quot; and return the standardized messages If applicable.
+- The client will process the response messages and depending on the content, either take no action or:
+  - Update the console with the list of errors or warnings
+  - Underline the erroneous area with swiggles and provide tips.
+
+The snapshot in figure (2) below provides an example of this experience.
+
+> ![](./screencap1.png)
+
+Figure (2) on the fly validation experience (taken from the current linter extension)
+
+2. Long running validation: a good example of this type of tools would be the live testing swagger validator. This tool picks up all sample request &amp; response payloads for the operations defined in the swagger, runs the sample request against the service and finally matches &amp; validates the response payload returned from the service to the sample response payload provided in the swagger.
+
+Running this type of validation continuously and on the fly will be very expensive, so the experience that we&#39;re proposing here is:
+
+- The user creates a new swagger or edits an existing one in VS code.
+- As the user updates the document no requests will be sent to Autorest to fire up long running validation tools.
+- When the user is ready to run a specific long running validation tools he will trigger it by:
+  - Selecting the JOSON file or giving it focus in VS Code.
+  - Clicking the tool&#39;s specific command button on the navigation as shown in figure (3) below, and yes these are **McDonalds** and **Burger King** Icons for now LOL (apologies this is the best of my photoshop skills ).
+  - Alternatively the user can right click on the JSON file in the file explorer and select the tool specific command as shown in figure (4) below.
+- The request will go to Autorest and the response with all the errors and warnings will be returned and displayed in the VS code console similar to the current behavior of the linter&#39;s extension. swiggles will also be displayed under the location of the violation with hints displayed when the users hovers over them.
+
+> ![](./screencap2.png)
+
+ (Figure 3) Trigger long running validations through navigation bar commands
+
+> ![](./screencap3.png)
+
+Figure (4) trigger long running operations through the file explorer&#39;s context menu

```

Comment:
@amarzavery  and I chatted quickly today about reviewing-- if we want to make a github code review extension, I'm all for that, but since there's no reason that they need to be 1 extension, we should probably make two if that's the case. (since, the PR feedback and swagger stuff will work perfectly independently anyway).